File: performance-investigation/fib/fib.c

#include <stdio.h>
#include <stdlib.h>

int fibonacci(int n) {
    if (n <= 1) {
        return n;
    }
    return fibonacci(n - 1) + fibonacci(n - 2);
}

int main(int argc, char *argv[]) {
    if (argc != 2) {
        printf("Usage: %s <n>\n", argv[0]);
        return 1;
    }

    int n = atoi(argv[1]);
    if (n < 0) {
        printf("n must be a non-negative integer.\n");
        return 1;
    }

    int result = fibonacci(n);
    printf("The %dth number in the Fibonacci sequence is: %d\n", n, result);

    return 0;
}

File: performance-investigation/fib/fib.java

public class fib {
    public static int fibonacci(int n) {
        if (n <= 1) {
            return n;
        }
        return fibonacci(n - 1) + fibonacci(n - 2);
    }

    public static void main(String[] args) {
        if (args.length != 1) {
            System.out.println("Usage: java Fibonacci <n>");
            System.exit(1);
        }

        int n = Integer.parseInt(args[0]);
        if (n < 0) {
            System.out.println("n must be a non-negative integer.");
            System.exit(1);
        }

        int result = fibonacci(n);
        System.out.println("The " + n + "th number in the Fibonacci sequence is: " + result);
    }
}
